### PR TITLE
add cluster table

### PR DESCRIPTION
**Pull request by daviszhen**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6662 

## What this PR does / why we need it:
Add cluster table that the sys account can read & write, the non-sys can only read.

the elementary design docs :https://doc.weixin.qq.com/doc/w3_AAgAkgYHABUbX2hx1RPRI2lSHAxw7?scode=AJsA6gc3AA881d74LvAAgAkgYHABU

### CODE REVIEW

**Review by reusee — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

File: pkg/container/vector/vector.go [hunk 1]
```diff
@@ -438,6 +438,59 @@ func (v *Vector) ConstExpand(m *mpool.MPool) *Vector {
 	return v
 }
 
+func (v *Vector) ConstExpandColAndNullsToFixedLength(m *mpool.MPool) *Vector {
```
Comment (nnsgmsone) [inline pkg/container/vector/vector.go:441]: Why not just use the constExpand function?

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

File: pkg/container/vector/vector.go (hunk 1 above)
Comment (daviszhen) [reply pkg/container/vector/vector.go:441]: The ConstExpand has the following code. 
That, the ConstExpand will not extend the field `Col` of the vector when it is the constant NULL. 
But, the auto increment need to update the field `Col` of the vector when the row is NULL. 
Unfortunately, the field `Col` of the vector is NULL that is extended by the ConstExpand.

I think the ConstExpand has been used in many places. So, It is unsuitable to modify the ConstExpand itself.
And, I add the function ConstExpandColAndNullsToFixedLength.

```
if v.IsScalarNull() {
		vlen := uint64(v.ScalarLength())
		nulls.AddRange(v.Nsp, 0, vlen)
		return v
	}
```

Comment (nnsgmsone) [reply pkg/container/vector/vector.go:441]: Functions with similar functionality should not be added. The current vector is messy enough.

Comment (daviszhen) [reply pkg/container/vector/vector.go]: > Functions with similar functionality should not be added. The current vector is messy enough.

I have added an extra parameter in the ConstExpand and removed ConstExpandColAndNullsToFixedLength

**Review by ouyuanning — CHANGES_REQUESTED** [review]
add some copy code in deepcopy.go when you change plan.proto

File: proto/plan.proto [hunk 2]
```diff
@@ -849,6 +852,7 @@ message DropTable {
 	string database = 2;
 	string table 	= 3;
 	repeated string index_table_names = 4;
+	ClusterTable cluster_table = 5;
```
[the remark this answers is not in the document]
Comment (daviszhen) [reply proto/plan.proto:855]: I have added deepcopy for the cluster table in dropTable,truncateTable,Node,InsertValues,

**Review by heni02 — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**